controllers/front/listing/ManufacturerController.php

```diff
@@ -239,8 +239,10 @@ public function getTemplateVarPage()
     {
         $page = parent::getTemplateVarPage();
 
-        $page['body_classes']['manufacturer-id-' . $this->manufacturer->id] = true;
-        $page['body_classes']['manufacturer-' . $this->manufacturer->name] = true;
+        if (Validate::isLoadedObject($this->manufacturer) && $this->manufacturer->active && $this->manufacturer->isAssociatedToShop()) {
+            $page['body_classes']['manufacturer-id-' . $this->manufacturer->id] = true;
+            $page['body_classes']['manufacturer-' . $this->manufacturer->name] = true;
+        }
```

Comment:
it's already checked during the initialization (see `init` and `initContent`). 
The real issue here, is that you are calling the parent initialization before the check (see `ManufacturerController::initContent`) the `parent::initContent` must be called after the check. 
So I suggest to solve this problem at lowest level.

Reply:
Not sure it will solve anything. I am solving manufacturers listing, there will never be a single manufacturer assigned. 🤷‍♂️